### ISSUE TITLE
[FEATURE] Add ability to GET custom endpoint

### ISSUE THREAD

**Issue by rodrigoaguilera** [Kephson/php-salesforce-rest-api#7]
## Summary / Description

I am using the salesforce knowledge base endpoint that require a GET HTTP request to fetch data. Currently the custom endpoint is hardcoded to POST.

## Describe the solution you'd like

`customEndpoint` accepts a new parameter `$method` which defaults to POST.

## Describe alternatives you've considered

I will just patch the code for my use case.

## Possible solution

```php
public function customEndpoint($customEndpoint, $data, $successStatusCode = 200, $additionalHeaders = [], $method = 'POST')
    {
        /* customEndpoint could be all behind /services/ */
        $url = "{$this->instanceUrl}/services/{$customEndpoint}";

        $headers = $this->getHeaders(
            [
                'Authorization' => "OAuth {$this->accessToken}",
                'Content-type' => 'application/json',
            ],
            $additionalHeaders
        );

        $client = new Client();

        try {
            $request = $client->request(
                $method,
                $url,
                [
                    'headers' => $headers,
                    'json' => $data
                ]
            );

            $status = $request->getStatusCode();
        } catch (ClientException $e) {
            throw SalesforceException::fromClientException($e);
        }

```

**Comment by Kephson** [comment]
Hi, thanks for implementing that; I merged your pull request and will release a new version.

**Comment by Kephson** [comment]
closed with new release 1.1.3, thanks for the pull request: https://github.com/Kephson/php-salesforce-rest-api/releases/tag/1.1.3
